### PR TITLE
allow support for saml + 2-factor

### PR DESCRIPTION
**Pull request by chrismeyersfsu**
related to https://github.com/ansible/awx/issues/48

* python-social-auth has SOCIAL_AUTH_SAML_SECURITY_CONFIG, which is
forwarded to python-saml settings configuration. This commit exposes
SOCIAL_AUTH_SAML_SECURITY_CONFIG to configure tower in tower to allow users
to set requestedAuthnContext, which will disable the requesting of password
type auth from the idp. Thus, it's up to the idp to choose which auth to
use (i.e. 2-factor).

### CODE REVIEW

**Comment by wenottingham** [comment]
As part of this, should we switch the default to "let the IdP decide"?

**Comment by chrismeyersfsu** [comment]
Yep, done.

**Review by wwitzel3 — CHANGES_REQUESTED** [review]
I'd like to see some unit tests for the new field types.

File: awx/main/tests/unit/conf/test_fields.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import pytest
```
Comment (jangsutsr) [inline awx/main/tests/unit/conf/test_fields.py]: There's `awx/conf/tests/unit/` available for hosting Tower Configuration tests.

**Review by wwitzel3 — APPROVED** [review]

**Pull request merged**